bytecomp/symtable.ml

```diff
@@ -24,13 +24,54 @@ open Cmo_format
 
 module String = Misc.Stdlib.String
 
+module Compunit = struct
+  type t = compunit
+  module Set = Set.Make(struct type nonrec t = t let compare = compare end)
+  module Map = Map.Make(struct type nonrec t = t let compare = compare end)
+end
+
+let builtin_values = Predef.builtin_values
+
+module Predef = struct
+  type t = predef
+  module Set = Set.Make(struct type nonrec t = t let compare = compare end)
+  module Map = Map.Make(struct type nonrec t = t let compare = compare end)
+end
+
+module Global = struct
+  type t =
+    | Glob_compunit of compunit
+    | Glob_predef of predef
+
+  let name = function
+    | Glob_compunit (Compunit cu) -> cu
+    | Glob_predef (Predef_exn exn) -> exn
+
+  let quote s = "`" ^ s ^ "'"
+
+  let string_of_global = function
+    | Glob_compunit (Compunit cu) -> "compilation unit " ^ (quote cu)
+    | Glob_predef (Predef_exn exn) -> "predefined exception " ^ (quote exn)
+
+  let global_of_ident id =
+    let name = Ident.name id in
+    if (Ident.is_predef id)
+    then Glob_predef (Predef_exn name)
+    else if (Ident.global id)
+    then Glob_compunit (Compunit name)
+    else assert false
```

Comment:
Do we really wan't an `assert false` here ? It seems that invalid_arg or failwith would be more appropriate

Reply:
It seems even better to have two function:
```ocaml
val persistent: Ident.t -> t option
val of_ident: Global.t -> t option
```
and remove the separate precondition check in the current callers of `global_of_ident`.